src/main/java/stirling/software/SPDF/config/security/SecurityConfiguration.java

```diff
@@ -140,6 +157,33 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
                                             .authenticated())
                     .userDetailsService(userDetailsService)
                     .authenticationProvider(authenticationProvider());
+
+            // Handle OAUTH2 Logins
+            if (applicationProperties.getSecurity().getOAUTH2().getEnabled()) {
+
+                 http.oauth2Login( oauth2 -> oauth2
+                         .loginPage("/oauth2")
+                         /*
+                         This Custom handler is used to check if the OAUTH2 user trying to log in, already exists in the database.
+                         If user exists, login proceeds as usual. If user does not exist, then it is autocreated but only if 'OAUTH2AutoCreateUser'
+                         is set as true, else login fails with an error message advising the same.
+                          */
+                         .successHandler(new AuthenticationSuccessHandler() {
+                                @Override
+                                public void onAuthenticationSuccess(HttpServletRequest request, HttpServletResponse response,
+                                        Authentication authentication) throws ServletException , IOException{
+                                    OAuth2User oauthUser =  (OAuth2User) authentication.getPrincipal();
+                                    if (userService.processOAuth2PostLogin(oauthUser.getAttribute("email"), applicationProperties.getSecurity().getOAUTH2().getAutoCreateUser())) {
```

Comment:
Will email always be the username key, are there not usecases were OAuth passes an actually userID?

Reply:
Googling it seems that 'sub' might be better since emails can change, Are you able to test with this?

Reply:
Unfortunately, no there is no standard `userid` claim supported by providers. Some return `id` claim (which is an integer) , some return `sub` claim (which is a 128-bit UUID). But everyone always support and return the `email` claim.

Reply:
> it seems that 'sub' might be better since emails can change,

Yes, that's true, unfortunately `sub` the newer version of the unique-id claim. And not all providers support it. Some still use the old `id` claim